forge/ee/db/models/Pipeline.js

```diff
@@ -7,7 +7,15 @@ module.exports = {
     schema: {
         name: {
             type: DataTypes.STRING,
-            allowNull: false
+            allowNull: false,
+            validate: {
```

Comment:
Validation being handled at a model rather than controller level, is a pattern I would like us to start repeating throughout our codebase. @knolleary @Steve-Mcl @joepavitt

Reply:
@Pezmc  I have no argument but can I trouble you to provide a little context please? I am not 100% on sequelize models and controllers and how every single piece fits together. I am assuming you raised this as you have read some best practice and it would improve out code quality? 

Ta.

Reply:
Sure! Ultimately it is the _model_ that is being validated, having validations and constraints at the model (and database level) means we can be 100% confident that all of our models are meeting the validation.

Right now, we usually validate the user input at the controller level, but there's nothing stopping an "invalid" model from being created, it also leads to duplicate validation logic spread throughout the app (e.g. to check a name is provided). Centralising this validation reduces code duplication, while also making our models more reliable.

Happy to spin up a separate issue (or drop into a huddle) to discuss further! I might end up writing this up as part of a "best practices" doc for us (thoughts @ZJvandeWeg @knolleary?)

Reply:
Sure! Ultimately it is the _model_ that is being validated, having validations and constraints at the model (and database level) means we can be 100% confident that all of our models are meeting the validation.

Right now, we usually validate the user input at the controller level, but there's nothing stopping an "invalid" model from being created, it also leads to duplicate validation logic spread throughout the app (e.g. to check a name is provided). Centralising this validation reduces code duplication, while also making our models more reliable.

Happy to spin up a separate issue (or drop into a huddle) to discuss further! I might end up writing this up as part of a "best practices" doc for us (thoughts @ZJvandeWeg @knolleary?)

Reply:
Controller validation should be done for security purposes and user feedback.

Models are indeed the level to validate on a table scale. There's always going the be some overlap.

Our code base has fairly fat controllers sadly, and do too much heavy lifting. More logic should reside in the model layer for a higher quality code base.

This should be documented indeed in developer docs. Would be great to do to raise the quality of contributions